### DISCUSSION TITLE
PolicyAssignments module - why nonComplianceMessages parameter is not an array ?

### DISCUSSION THREAD

**Discussion by dibcar** [Azure/ResourceModules#1625] [Q&A]
How can I use the policyAssignments module for an policy initiative and configure a non-compliance message per policy? The nonComplanceMessages parameter is defined as a string instead of an array as defined in API version 2021-06-01.

Arm template : 
"nonComplianceMessages": [
    {
        "message": "Default message"
    },
    {
        "message": "Message for just this policy definition by reference ID",
        "policyDefinitionReferenceId": "10420126870854049575"
    }
]

**Comment by AlexanderSehr** [comment]
@ahmadabdalla , can you have a look?

**Reply by ahmadabdalla** [reply]
@dibcar @MrMCake will review this

**Reply by ahmadabdalla** [reply]
@dibcar thanks for raising this. This is a bug, and this PR https://github.com/Azure/ResourceModules/pull/1644 should provide the fix.
